lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -36,10 +36,16 @@ class GoogleCloudOutput < BufferedOutput
     config_param :private_key_path, :string, :default => nil
     config_param :private_key_passphrase, :string, :default => 'notasecret'
 
-    # If use_metadata_service is set to true, we obtain the project_id, zone,
-    # and vm_id from the GCE metadata service.  Otherwise, those parameters
-    # must be specified in the config file explicitly.
+    # Specify project/instance metadata.
+    #
+    # project_id, zone, and vm_id are required to have valid values, which
+    # can be obtained from the metadata service or set explicitly.
+    # Otherwise, the plugin will fail to initialize.
+    #
+    # Whether to attempt to obtain metadata from the local metadata service.
+    # It is safe to specify 'true' even on platforms with no metadata service.
     config_param :use_metadata_service, :bool, :default => true
```

Comment:
It won't be all or nothing. The specific scenario I'm referring to is as follows:
User wants to override e.g. zone, but he does some mistake/forgets to do it.
If the flag defaults to true, it will be fetched automatically and logs will have wrong data.

The alternative solution would be something like:
if project_ud or zone or vm_id then use_metadata_service, default => false
else use_metadata_service, default => true

However I don't know how likely is that scenario, you probably have a better idea if it's worth pursuing that route :)